tests/algorithms/test_connected.py

```diff
@@ -0,0 +1,66 @@
+import pytest
+from tests.conftest import edgelist1
+import xgi
+from xgi.exception import XGIError
+from xgi.readwrite import edgelist
+
+
+def test_is_connected(edgelist1, edgelist3):
+    # edgelist1 = [[1,2,3],[4],[5,6],[6,7,8]]
+    # edgelist2 = [[1,2,3],[3,4],[4,5,6]]
+    el1 = edgelist1
+    el2 = edgelist3
+    H1 = xgi.Hypergraph(el1)
+    H2 = xgi.Hypergraph(el2)
+    assert xgi.is_connected(H1) == False
+    assert xgi.is_connected(H2) == True
+
+
+def test_connected_components(edgelist1, edgelist2, edgelist4):
+    el1 = edgelist1
+    el2 = edgelist2
+    el4 = edgelist4
+    H1 = xgi.Hypergraph(el1)
+    H2 = xgi.Hypergraph(el2)
+    H3 = xgi.Hypergraph(el4)
+    assert sorted([len(cc) for cc in xgi.connected_components(H1)]) == [1, 3, 4]
+    assert {1, 2, 3} in xgi.connected_components(H1)
+    assert {4} in xgi.connected_components(H1)
+    assert {5, 6, 7, 8} in xgi.connected_components(H1)
+    assert sorted([len(cc) for cc in xgi.connected_components(H2)]) == [2, 4]
+    assert {1, 2} in xgi.connected_components(H2)
+    assert {3, 4, 5, 6} in xgi.connected_components(H2)
+    assert sorted([len(cc) for cc in xgi.connected_components(H3)]) == [5]
+    assert {1, 2, 3, 4, 5} in xgi.connected_components(H3)
```

Comment:
Suggestion: this is both faster and shorter.
```suggestion
    cc1 = xgi.connected_components(H1)
    cc2 = xgi.connected_components(H2)
    cc3 = xgi.connected_components(H3)
    assert sorted([len(cc) for cc in cc1]) == [1, 3, 4]
    assert {1, 2, 3} in cc1
    assert {4} in cc1
    assert {5, 6, 7, 8} in cc1
    assert sorted([len(cc) for cc in cc2]) == [2, 4]
    assert {1, 2} in cc2
    assert {3, 4, 5, 6} in cc2
    assert sorted([len(cc) for cc in cc3]) == [5]
    assert {1, 2, 3, 4, 5} in cc3
```

Reply:
Fixed!